support/api-testsupport/src/main/java/com/hotels/styx/support/api/BlockingObservables.java

```diff
@@ -25,6 +25,13 @@
         return observable.toBlocking().single();
     }
 
+    public static FullHttpResponse waitForSingleResponse(Observable<HttpResponse> responseObs) {
+        return responseObs
+                .flatMap(response -> response.toFullResponse(120*1024))
+                .toBlocking()
+                .single();
+    }
+
```

Comment:
Please consider fixing the original `waitForesponse` method. Did you try modifying that to use `.singe()` instead of `first()`?

Reply:
Yes, tests fail.